### PR TITLE
Inference using PINTS

### PR DESCRIPTION
**Pull request by ben18785**
Currently:

- fixes noise parameters at user-specified values
- uses the default parameter values for the optimiser or sampler starting values
- is untested for multiple output inference problems (this was because I was unable to figure out whether I could use one of the multiple output pk models in this way). Note the Myokit forward model wrapping _is_ tested -- just the PINTS inference is not
- creates an `InferenceFunctionResults` model that captures function evaluations for each iteration for each chain

### CODE REVIEW

**Comment by ben18785** [comment]
Closing as thing runners have hung due to this line `run_inference.delay(new_inference.id)`.

File: pkpdapp/pkpdapp/tests/test_models/test_inference_mixin.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        # create mixin object
+        self.inference_mixin = InferenceMixin(self.inference)
+
+    def test_objective_functions(self):
+        # Test that log-likelihood, log-prior and log-posterior work
+
+        # Test log-likelihood
+        log_likelihood = self.inference_mixin.create_pints_log_likelihood()
+        val = log_likelihood([1, 1, 1, 1, 1])
+        self.assertTrue(abs(val - -113.33099855566624) < 0.1)
```
Comment (martinjrobins) [inline pkpdapp/pkpdapp/tests/test_models/test_inference_mixin.py:79]: use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual

Comment (martinjrobins) [reply pkpdapp/pkpdapp/tests/test_models/test_inference_mixin.py:79]: and below

**Review by martinjrobins — CHANGES_REQUESTED** [review]
this is great, thanks @ben18785. I've noted a few changes, and I'll try to deal with these myself today

File: pkpdapp/pkpdapp/models/myokit_pints_forward_model.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    Creates a `pints.ForwardModel`.
+    Arguments:
+        myokit_model -- a Myokit model.
+        myokit_simulator -- a Myokit simulator.
+        outputs(=None by default) -- a list of strings
+        representing state names in model
+        fixed_parameter_dict(=None by default) -- a dictionary
+        representing key-value pairs for fixed parameters
+    """
+    def __init__(self, myokit_model, myokit_simulator, outputs=None,
```
[the remark this answers is not in the document]
Comment (martinjrobins) [reply pkpdapp/pkpdapp/models/myokit_pints_forward_model.py:22]: actually you can't!

**Review by martinjrobins — APPROVED** [review]
happy to merge now

**Pull request merged**